src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -411,7 +412,7 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     completionWork onFailure {
       case e =>
         logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
-        scheduleTransition(WorkflowFailed)
+        self ! CheckForWorkflowComplete
```

Comment:
Hmm, I'm on the edge here. One thing I'd say - at the moment all of the functions make sure they leave database state consistent when the actor is "between" the processing of messages. They mess around in all sorts of ways "during" the processing of messages.

So, if I want to make sure checkForWorkflowComplete looks at a complete and consistent view of the world (which I do), I want to make it happen outside of any other message processing activity. There's a lot less thinking about "is it appropriate to check this now with the current state of the state" required if it's isolated in its own message. And I *hate* thinking.